stan/math/prim/scal/prob/normal_rng.hpp

```diff
@@ -13,26 +17,42 @@
 namespace stan {
   namespace math {
 
-    template <class RNG>
-    inline double
-    normal_rng(double mu,
-               double sigma,
+    template <typename T_loc, typename T_scale, class RNG>
+    inline typename promote_vector<T_loc, T_scale>::type
+    normal_rng(const T_loc &mu,
+               const T_scale &sigma,
                RNG& rng) {
       using boost::variate_generator;
       using boost::normal_distribution;
 
       static const char* function("normal_rng");
 
+      scalar_seq_view<T_loc> mu_vec(mu);
+      scalar_seq_view<T_scale> sigma_vec(sigma);
+
       check_finite(function, "Location parameter", mu);
       check_not_nan(function, "Location parameter", mu);
       check_positive(function, "Scale parameter", sigma);
       check_not_nan(function, "Scale parameter", sigma);
+      if (mu_vec.size() > 1 && sigma_vec.size() > 1)
```

Comment:
This isn't the right test.  What if a vector is of size zero?  What you need to do is check for compatibility in cases where you have a container.  That's what's going on in the distributions, so you use the same logic and call `check_consistent_sizes`.